app/src/main/java/com/jalasoft/routesapp/ui/auth/login/viewModel/LoginViewModel.kt

```diff
@@ -0,0 +1,169 @@
+package com.jalasoft.routesapp.ui.auth.login.viewModel
+
+import android.annotation.SuppressLint
+import android.app.AlertDialog
+import android.content.Context
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.google.firebase.auth.AuthCredential
+import com.jalasoft.routesapp.R
+import com.jalasoft.routesapp.data.remote.managers.UserRepository
+import com.jalasoft.routesapp.util.helpers.UserTypeLogin
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel
+@Inject
+constructor(private val repository: UserRepository) : ViewModel() {
+
+    val loginIsSuccessful: MutableLiveData<Boolean> by lazy {
+        MutableLiveData<Boolean>()
+    }
+    val errorMessage: MutableLiveData<String> by lazy {
+        MutableLiveData<String>()
+    }
+    val signInGoogle: MutableLiveData<Boolean> by lazy {
+        MutableLiveData<Boolean>()
+    }
+
+    @SuppressLint("StaticFieldLeak")
+    var context: Context? = null

```

Comment:
check the alert comment below, and once you're done with it remove the context

Reply:
i did not remove this context because this is from Adriana's merge i only remove the one from my RegisterUserViewModel